src/bundle.ts

```diff
@@ -135,7 +136,7 @@ async function bundle(
     ? (await fileExists(entryPath)) && (await fs.stat(entryPath)).isFile()
     : false
 
-  if (!hasSpecifiedEntryFile && !isMultiEntries) {
+  if (!hasSpecifiedEntryFile && !isMultiEntries && !hasBin) {
```

Comment:
let's merge the hasBin and isMultiEntries flag, since we're treatign binary as one of the entries